network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -2008,6 +2038,98 @@ private <T extends IdentifiableAttributes> void deletePermanentLimits(UUID netwo
         resourceIdsByVariant.forEach((k, v) -> deletePermanentLimits(networkUuid, k, v));
     }
 
+    // Regulation Points
+    public void insertRegulationPoints(Map<OwnerInfo, RegulationPointAttributes> regulationPoints) {
+        try (var connection = dataSource.getConnection()) {
+            try (var preparedStmt = connection.prepareStatement(QueryCatalog.buildInsertRegulationPointsQuery())) {
+                List<Object> values = new ArrayList<>(8);
+                List<Map.Entry<OwnerInfo, RegulationPointAttributes>> list = new ArrayList<>(regulationPoints.entrySet());
+                for (List<Map.Entry<OwnerInfo, RegulationPointAttributes>> subUnit : Lists.partition(list, BATCH_SIZE)) {
+                    for (Map.Entry<OwnerInfo, RegulationPointAttributes> attributes : subUnit) {
+                        values.clear();
+                        values.add(attributes.getKey().getNetworkUuid());
+                        values.add(attributes.getKey().getVariantNum());
+                        values.add(attributes.getValue().getRegulatedEquipmentId());
+                        values.add(attributes.getValue().getRegulationMode());
+                        if (attributes.getValue().getLocalTerminal() != null) {
```

Comment:
values.add(attributes.getValue().getLocalTerminal() != null 
   ? attributes.getValue().getLocalTerminal().getConnectableId() 
   : null);
values.add(attributes.getValue().getLocalTerminal() != null 
  ? attributes.getValue().getLocalTerminal().getSide() 
  : null);

You should add null value if attributes.getValue().getLocalTerminal() is null